Earthfile

```diff
@@ -11,24 +11,36 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-VERSION 0.6
-FROM ubuntu:jammy
+VERSION 0.8
 
-# Defaulting to ROS 2 humble
-ARG ROS_DISTRO="humble"
+ARG --global IMAGE_NAME="osrf/space-ros"
+ARG --global USERNAME="spaceros-user"
+ARG --global HOME="/home/${USERNAME}"
+ARG --global WORKSPACE_DIR="/workspace"
+ARG --global TAG=""  # See discussion: #140
 
-setup:
-  # Disable prompting during package installation
-  ARG DEBIAN_FRONTEND=noninteractive
-  ARG ROS_DISTRO
+all:
+  BUILD +base-image
+  BUILD +core-image
+  BUILD +dev-image
 
-  # TODO - the `setup` step will be merged with the `setup` step in spaceros docker Earthfile
-  # This variable will then act as a single source of truth.
-  ENV ROS_DISTRO ${ROS_DISTRO}
+core-image:
+  BUILD +image --IMAGE_VARIANT=core
 
-  # The following commands are based on the source install for ROS 2 Rolling Ridley.
-  # See: https://docs.ros.org/en/ros2_documentation/rolling/Installation/Ubuntu-Development-Setup.html
-  # The main variation is getting Space ROS sources instead of the Rolling sources.
+base-image:
+  BUILD +image --IMAGE_VARIANT=base
+
+dev-image:
+  BUILD +image --IMAGE_VARIANT=dev
+
+pre-installation:
+  FROM ubuntu:jammy
+
+  ENV DEBIAN_FRONTEND=noninteractive
+  ENV ROSDISTRO="humble"
```

Comment:
i thought we just changed it to ROS_DISTRO?

Reply:
I always had this confusion. Will fix it.